Black: new port

danchr opened the pull request:
#### Description

<!-- Note: it is best make pull requests from a branch rather than from master -->
This adds a port for the Black code formatter for Python. Since it requires `py-toml`, I updated its list of supported pythons so that Black could run in Python 3.6 as well.


###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
<!-- Generate version information with this command in shell:
    echo "macOS $(sw_vers -productVersion) $(sw_vers -buildVersion)"; echo "Xcode $(xcodebuild -version | awk '{print $NF}' | tr '\n' ' ')"
-->
macOS 10.x
Xcode 8.x

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?
<!-- Please don't open a new Trac ticket if you are submitting a pull request. -->
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

mf2k commented on python/py-black/Portfile:
```diff
@@ -0,0 +1,45 @@
+        port:py${python.version}-click \
+        port:py${python.version}-attrs \
+        port:py${python.version}-appdirs \
+        port:py${python.version}-toml
```
Please sort the library dependencies alphabetically.

tobypeterson commented:
Why support older pythons? I suppose 36 is vaguely acceptable, but definitely no reason to add any of the older versions to py-toml

danchr commented:
Why not? They are supported both upstream and by MacPorts elsewhere; why is
`py-toml` special in this regard?

man. 25. mar. 2019 kl. 17.49 skrev tobypeterson <notifications@github.com>:

> Why support older pythons? I suppose 36 is vaguely acceptable, but
> definitely no reason to add any of the older versions to py-toml
>
> —
> You are receiving this because you authored the thread.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/macports/macports-ports/pull/3934#issuecomment-476284510>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AADnpKp-St3K8VWWD8rPyR-DQobTavirks5vaP4VgaJpZM4cGm6z>
> .
>
-- 
Dan Villiom Podlaski Christiansen
danchr@gmail.com

tobypeterson commented:
I prefer to add ports that are needed, rather than merely supported. Black needs it for python 3.6 - fine. Nothing needs it for 2.x or older 3.x releases.

mf2k commented:
I generally agree with Toby, let's just support the latest python 2 and 3. Ultimately more python sub-port means more maintenance in the form of testing and also more builder time used. However, if you are willing to test all of those versions every upgrade, then by all means go ahead.

mf2k commented:
Dan, the reason there are other ports with old python versions is simply because those were the only ones available when the port was added. As new pythons came out, we added sub-ports. There is really no reason to go back and add old python sub-ports.

danchr commented:
> Dan, the reason there are other ports with old python versions is simply because those were the only ones available when the port was added. As new pythons came out, we added sub-ports. There is really no reason to go back and add old python sub-ports.

Well, I mostly disagree; having the supported Pythons around is quite useful to me. In this case, though, given that `black` is mostly a tool, there isn't much point in supporting multiple Python versions anyway, so I changed it to just always use 3.7.

mf2k commented:
Thank you!
